### PR TITLE
Cascade host fees changes to children

### PR DESCRIPTION
**Pull request by Betree**
Resolve https://github.com/opencollective/opencollective/issues/5065

- Cascade host fee changes to children
- Add a script to update host fees for existing transactions

Usage:

```bash
$ npm run script scripts/ledger/update-transactions.ts -- --account apex --hostFeePercent 15
648 transaction(s) to update with the following infos:
 - Update 324 transactions with 15% host fee

This action is irreversible. Are you sure you want to continue? (Yes/No)
> 
```

# Migration strategy for the Social Change Nest

1. Deploy
2. Run the script
3. Manually update the fee structure and `hostFeePercent` for all concerned accounts (see the slugs list below)
4. Check the budgets to make sure everything is correct

> 'gaf-film-club-discusses-social-reproduction-and-solidarity-economy-53e79758','connecting-with-health-professionals-58e2cba9','calling-ethnic-minority-applied-psychologists-32cff7cb','trust-the-people-online-course-0160501e','global-poetry-showcase','vappaw-future','reclaiming-our-town-centres-7c343779','community-family-christmas-party-c9e5e5cb','cylch-merched-launch-party-e5144e59'

SQL query:
```sql
SELECT c."hostFeePercent", pc."hostFeePercent", pc."data", c."data" 
FROM "Collectives" c 
INNER JOIN "Collectives" pc ON c."ParentCollectiveId" = pc.id
WHERE c."HostCollectiveId" = 98478
AND c."isActive" IS TRUE
AND c."hostFeePercent" != pc."hostFeePercent" 
AND c."deletedAt" IS NULL
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -137,6 +137,7 @@
     "chai-as-promised": "^7.1.1",
     "chai-jest-snapshot": "^2.0.0",
     "codecov": "^3.8.2",
+    "commander": "^8.3.0",
```
Comment (Betree) [inline package.json:140]: Introduced this one to have an easy way to parse command-line args

Comment (znarf) [reply package.json:140]: We used to have `argparse` in some scripts. Would be great to consolidate.

Maybe a small RFC to explain the choice?

File: scripts/ledger/update-transactions.ts [hunk 2]
```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env ./node_modules/.bin/babel-node
+
+/**
+ * This script aims to provide a command-line interface to update transactions in a safe way.
+ */
+
+import '../../server/env';
+
+import { Command } from 'commander'; // eslint-disable-line node/no-unpublished-import
```
Comment (Betree) [inline scripts/ledger/update-transactions.ts:9]: Let's switch to `argparse`

**Comment by Betree** [comment]
I initially missed that we had `argparse` available in the codebase, I should have used that from the start. However when I tried to migrate my work there I realized that - in comparison - `commander` actually provides a very friendly API with better documentation and proper maintenance. I'll therefore go with the comment in https://github.com/opencollective/opencollective-api/pull/7003#discussion_r777340020 suggesting to open an RFC to consolidate things.

**Pull request merged**